test/testUtils.ts

```diff
@@ -0,0 +1,9 @@
+import t from 'tap'
+
+export async function throwsWithMessage(funcUnderTest: () => {}, expectedMessage: string) {
```

Comment:
I _think_ this should take a `t` as a parameter and not use the global `t` instance. When you're using sub-tests `tap` can be confused if you use the wrong `t`.

Reply:
It originally took the tap instance as a parameter but I didn't see any use in keeping it.
That's a good point about the sub-tests though, it might mess up reporting the failures. I'll add it back in.